### PR TITLE
Add BinaryBuilder binaries with fallback to existing source build

### LINKED ISSUE #13: Building Error in Windows
When I try to build the GLMNet.jl it tells me that the file is not found. I dug around and it looks like the problem is inside the build.jl file:

``` javascript
pic = @windows ? "" : "-fPIC"
run(`gfortran -m$WORD_SIZE -fdefault-real-8 -ffixed-form $pic -shared -O3 glmnet5.f90 -o libglmnet.so`)
```

and when the `run( )` line is read, under windows it will change `$pic` to `''` instead of an actual blank, which messes up the build. When I removed the `$pic` from that line, it built fine

### PR DESCRIPTION
**Pull request by JackDunnNZ**
I am trying to deploy with a dependency on GLMNet in a setting where it's unreasonable to ask Windows users to have a fortran compiler, so I wanted to add the option of pre-built binaries with a fallback to the existing source build, just in case.

This also enables testing windows on appveyor and osx on travis. I'm guessing appveyor needs to be enabled for the main repo, but it is passing on my fork:
https://ci.appveyor.com/project/JackDunnNZ/glmnet-jl

The builder repo is https://github.com/JackDunnNZ/GLMNetBuilder, I am happy to transfer to JuliaStats (or anyone else) if desired. 

Fixes #13

### CODE REVIEW

**Comment by JackDunnNZ** [comment]
Bump, would appreciate it if someone from @JuliaStats could look over this

File: deps/build.jl [hunk 1]
```diff
@@ -1,10 +1,106 @@
-using Compat
-using Libdl 
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
```
Comment (simonbyrne) [inline deps/build.jl]: Should this be added to the REQUIRE file?

Comment (simonbyrne) [reply deps/build.jl]: (it seems to work without it, but I assume that's because it's installed by some dependant library)

Comment (ararslan) [reply deps/build.jl]: Yes, it needs to be in REQUIRE.

**Comment by simonbyrne** [comment]
Thanks, sorry about the delay.

**Comment by JackDunnNZ** [comment]
Thanks!

**Comment by JackDunnNZ** [comment]
@simonbyrne would it be possible to tag a new release?

**Comment by simonbyrne** [comment]
done!

**Comment by JackDunnNZ** [comment]
Thanks so much!

**Pull request merged**